go/client/logclient.go

```diff
@@ -29,8 +29,18 @@ const (
 	GetEntriesPath  = "/ct/v1/get-entries"
 )
 
-// LogClient represents a client for a given CT Log instance
-type LogClient struct {
+// LogClient defines an interface for Log Clients.  This aids testing.
+type LogClient interface {
+	AddChain(chain []ct.ASN1Cert) (*ct.SignedCertificateTimestamp, error)
+	AddPreChain(chain []ct.ASN1Cert) (*ct.SignedCertificateTimestamp, error)
+	AddChainWithContext(ctx context.Context, chain []ct.ASN1Cert) (*ct.SignedCertificateTimestamp, error)
+	AddJSON(data interface{}) (*ct.SignedCertificateTimestamp, error)
+	GetSTH() (sth *ct.SignedTreeHead, err error)
+	GetEntries(start, end int64) ([]ct.LogEntry, error)
+}
+
+// logClientImpl conforms to LogClient and represents a client for a given CT Log instance
+type logClientImpl struct {
```

Comment:
I'm not massively in favour of having type names that differ only the case of a single character.